### PR TITLE
Feat/KFS-1768 KFS-1850 improve flink shell extractability

### PR DESCRIPTION
**Pull request by Guttz**
Release Notes
-------------

New Features
- Added new SET option 'client.output-format' to `confluent flink shell` supporting plain text mode.

Checklist
---------
- [x] Leave this box unchecked if features are not yet available in production

What
----

References
----------

Test & Review
-------------

yes

### CODE REVIEW

File: pkg/flink/config/local_statements.go [hunk 1]
```diff
@@ -21,6 +21,15 @@ const (
 	KeyResultsTimeout = "client.results-timeout"
 	KeyServiceAccount = "client.service-account"
 	KeyStatementName  = "client.statement-name"
+	KeyOutputFormat   = "client.output-format"
```
Comment (YannickPferr) [inline pkg/flink/config/local_statements.go:24]: Could we add auto completions for this in language-service?

File: pkg/flink/internal/controller/interactive_output_controller.go [hunk 2]
```diff
@@ -22,14 +23,16 @@ type InteractiveOutputController struct {
 	tableView     components.TableViewInterface
 	resultFetcher types.ResultFetcherInterface
 	isRowViewOpen bool
+	outputFormat  func() config.OutputFormat
 	debug         bool
 }
 
-func NewInteractiveOutputController(tableView components.TableViewInterface, resultFetcher types.ResultFetcherInterface, debug bool) types.OutputControllerInterface {
+func NewInteractiveOutputController(tableView components.TableViewInterface, resultFetcher types.ResultFetcherInterface, outputFormat func() config.OutputFormat, debug bool) types.OutputControllerInterface {
```
Comment (YannickPferr) [inline pkg/flink/internal/controller/interactive_output_controller.go:30]: I don't really like that we have to pass functions around `outputFormat func() config.OutputFormat`. I think we should instead extract the UserProperties out of the store and then pass those directly. Wdyt?
You don't have to do it now if it's to big of a refactor though

File: pkg/flink/config/local_statements.go [hunk 3]
```diff
@@ -21,6 +21,15 @@ const (
...
+)
+
+type OutputFormat string
+
+const (
+	OutputFormatStandard  OutputFormat = "standard"
+	OutputFormatPlainText OutputFormat = "plain-text"
+	//OutputFormatJson  = "json"
```
Comment (brianstrauch) [inline pkg/flink/config/local_statements.go:32]:
```suggestion
```

File: pkg/flink/internal/controller/interactive_output_controller.go [hunk 4]
```diff
@@ -192,6 +198,14 @@ func (t *InteractiveOutputController) handleKeyUpOrDownPress(event *tcell.EventK
 	return event
 }
 
+func (t *InteractiveOutputController) withBorder() bool {
+	if t.outputFormat() == config.OutputFormatPlainText {
+		return false
+	} else {
+		return true
+	}
```
[the remark this answers is not in the document]
Comment (Guttz) [reply pkg/flink/internal/controller/interactive_output_controller.go:206]: done

File: pkg/flink/internal/controller/base_output_controller.go [hunk 5]
```diff
@@ -5,28 +5,31 @@ import (
...
 	"github.com/olekukonko/tablewriter"
 
+	"github.com/confluentinc/cli/v3/pkg/flink/config"
 	"github.com/confluentinc/cli/v3/pkg/flink/internal/results"
 	"github.com/confluentinc/cli/v3/pkg/flink/internal/utils"
 	"github.com/confluentinc/cli/v3/pkg/flink/types"
 )
 
-type BasicOutputController struct {
+type BaseOutputController struct {
```
[the remark this answers is not in the document]
Comment (Guttz) [reply pkg/flink/internal/controller/base_output_controller.go:14]: I'll maintain only BaseOutputController

File: pkg/flink/internal/controller/interactive_output_controller.go (hunk 2 above)
Comment (Guttz) [reply pkg/flink/internal/controller/interactive_output_controller.go]: Some refactoring was involved, but done

File: pkg/flink/internal/store/store.go [hunk 6]
```diff
@@ -325,38 +313,16 @@ func extractPageToken(nextUrl string) (string, error) {
...
-		Properties:       NewUserProperties(getDefaultProperties(appOptions), getInitialProperties(appOptions)),
+		Properties:       userProperties,
 		client:           client,
 		exitApplication:  exitApplication,
 		appOptions:       appOptions,
 		tokenRefreshFunc: tokenRefreshFunc,
 	}
 }
 
-func getDefaultProperties(appOptions *types.ApplicationOptions) map[string]string {
```
Comment (YannickPferr) [inline pkg/flink/internal/store/store.go:334]: nice

**Review by YannickPferr — APPROVED** [review]

File: pkg/flink/internal/controller/interactive_output_controller.go (hunk 2 above)
Comment (YannickPferr) [reply pkg/flink/internal/controller/interactive_output_controller.go]: very cool, thanks!

File: test/fixtures/output/flink/shell/reset-all-keys.golden [hunk 7]
```diff
@@ -26,6 +26,7 @@ configuration has been reset successfully.
 +------------------------+---------------------+
 |          Key           |        Value        |
 +------------------------+---------------------+
+| client.output-format   | standard (default)  |
```
Comment (brianstrauch) [inline test/fixtures/output/flink/shell/reset-all-keys.golden:29]: Add `test/fixures/output/flink/shell` to CODEOWNERS so you don't need to ask for a review?

**Review by brianstrauch — APPROVED** [review]

**Pull request merged**